ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -937,6 +982,53 @@ func checkK8sJobAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJo
 	return false
 }
 
+func (r *RayJobReconciler) checkSidecarContainerAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJob) bool {
+	logger := ctrl.LoggerFrom(ctx)
+
+	rayClusterInstance, err := r.getOrCreateRayClusterInstance(ctx, rayJob)
+	if err != nil {
+		logger.Error(err, "Failed to get RayCluster instance for checking sidecar container status")
+		return false
+	}
+
+	headPod, err := common.GetRayClusterHeadPod(ctx, r.Client, rayClusterInstance)
+	if err != nil {
+		logger.Error(err, "Failed to get Ray head pod for checking sidecar container status")
+		return false
+	}
+
+	if headPod == nil {
+		logger.Info("Ray head pod not found, skipping sidecar container status check")
+		return false
+	}
+
+	// Check container statuses for any error conditions
+	for _, containerStatus := range headPod.Status.ContainerStatuses {
+		// Check for terminated containers with error exit codes
+		if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode != 0 {
```

Comment:
We should only check the submitter container instead of all containers. For example, it's common for users to inject sidecar containers like FluentBit to collect logs, and the RayJob CR will be marked as `AppFailed` when the FluentBit container failed which is misleading.